src/metrics/promotions.ts

```diff
@@ -33,61 +34,94 @@ export class PromotionMetric implements Metric {
-        
+
         // increment queen count if a promotion occurs
-        if (move.promotion === "q") {
-          thisGameQueenCount++;
+        if (move.promotion === 'q') {
+          currentQueens++;
+          totalQueens++;
+          // the check below is only necessary on promotions, when queen count might have increased beyond the max
+
+          // identify the maxQueenCount in a particular move, and the games and moves that the maxQueenCount occured in
+          // push to array of games and moves if tie, otherwise wipe the array and add new game
+          // only add one move entry for each game maxQueenCount (rather than one entry for each move that the maxQueenCount appears in)
+          if (currentQueens > this.maxQueensOnBoard) {
+            this.maxQueensOnBoard = currentQueens;
+            this.movesAndGamesWithMaxQueensOnBoard = [
+              {
+                game: gameSite,
+                move: move.originalString,
+              },
+            ];
+          } else if (currentQueens === this.maxQueensOnBoard) {
+            if (
+              !this.movesAndGamesWithMaxQueensOnBoard.some(
+                (item) => item.game === gameSite
+              )
+            ) {
+              this.movesAndGamesWithMaxQueensOnBoard.push({
+                game: gameSite,
+                move: move.originalString,
+              });
+            }
+          }
         }
       }
 
       // decrement queen count if queen is captured
-      if (move.capture && (move.capture.uas === "q" || move.capture.uas === "Q")) {
-        thisGameQueenCount--;
+      if (
+        move.capture &&
+        (move.capture.uas === 'q' || move.capture.uas === 'Q')
+      ) {
+        currentQueens--;
       }
+    }
 
-      // identify the maxQueenCount in a particular move, and the games and moves that the maxQueenCount occured in
-      // push to array of games and moves if tie, otherwise wipe the array and add new game
-      // only add one move entry for each game maxQueenCount (rather than one entry for each move that the maxQueenCount appears in)
-      if (thisGameQueenCount > this.maxQueenCounts) {
-        this.maxQueenCounts = thisGameQueenCount;
-        this.movesAndGamesWithMaxQueenCount = [{
+    // after the game is done, update the maxQueens property and store the game it occured in
+    if (totalQueens > this.maxQueens) {
+      this.maxQueens = totalQueens;
+      this.movesAndGamesWithMaxQueens = [
+        {
           game: gameSite,
-          move: move.originalString,
-        }];
-      } else if (thisGameQueenCount === this.maxQueenCounts) {
-        if (!this.movesAndGamesWithMaxQueenCount.some((item) => item.game === gameSite)) {
-          this.movesAndGamesWithMaxQueenCount.push({
-            game: gameSite,
-            move: move.originalString
-          });
-        }
+          move: game[0].move.originalString,
+        },
+      ];
+    } else if (totalQueens === this.maxQueens) {
```

Comment:
so here we are trying to find the games that have had the most unique queens appear on board total, rather than the move/game with the most queens in a given time? nice

Reply:
yep